src/github.com/couchbase/sync_gateway/db/change_cache_test.go

```diff
@@ -689,6 +688,38 @@ func FailingTestChannelRace(t *testing.T) {
 	close(options.Terminator)
 }
 
+// Test retrieval of skipped sequence using view.  Unit test catches panic, but we don't currently have a way
+// to simulate an entry that makes it to the bucket (and so is accessible to the view),
+// but doesn't show up on the TAP feed.  Cache logging in this test validates that view retrieval is working, but
+// the subsequent addition of the item returned by the view call to the cache is treated as a duplicate, as it was already added
+// when it showed up on the tap feed.
+func TestSkippedViewRetrieval(t *testing.T) {
+	base.LogKeys["Cache"] = true
+	base.LogKeys["Cache+"] = true
+	db := setupTestDBWithCacheOptions(t, shortWaitCache())
+	defer tearDownTestDB(t, db)
+	db.ChannelMapper = channels.NewDefaultChannelMapper()
+
+	// Allow db to initialize and run initial CleanSkippedSequenceQueue
+	time.Sleep(10 * time.Millisecond)
+
+	// Write sequence 3 direct
+	WriteDirect(db, []string{"ABC"}, 3)
+
+	// Artificially add 3 and 5 to skipped, and back date skipped entry by 2 hours to trigger attempted view retrieval during Clean call
+	db.changeCache.skippedSeqs.Push(&SkippedSequence{3, time.Now().Add(time.Duration(time.Hour * -2))})
+	db.changeCache.skippedSeqs.Push(&SkippedSequence{5, time.Now().Add(time.Duration(time.Hour * -2))})
+	db.changeCache.CleanSkippedSequenceQueue()
+
+	// Validate that 3 is in the channel cache, 5 isn't
+	time.Sleep(10 * time.Millisecond)
```

Comment:
can you add a comment about why the time.Sleep() is needed?  (to give someone a hint on what they'd need to do to remove it some day)